src/input_handler.cc

```diff
@@ -821,6 +822,10 @@ class Prompt : public InputMode
             m_callback(line, PromptEvent::Abort, context());
             return;
         }
+        else if ((key == Key::Backspace or key == ctrl('h')) and line.empty())
```

Comment:
Merging this into line 810 condition should do the trick, `goto` is quite tricky in C++ because it cannot span any object lifetime starting/finishing, which adds constraints on how th code can be structured. As @krobelus said history_push will check that the line is empty (we dont want to add empty lines even if we press escape) so the behaviour should be fine.